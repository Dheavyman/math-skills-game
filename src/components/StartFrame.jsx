import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  handleStartGame: PropTypes.func.isRequired,
};

/**
 * Start frame component
 *
 * @param {object} props - Properties passed to component
 *
 * @returns {object} React element
 */
const StartFrame = (props) => {
  const { handleStartGame } = props;

  return (
    <div className="row mt-5">
      <div className="col-12 text-center">
        <button
          type="button"
          className="btn btn-lg btn-info"
          onClick={handleStartGame}
        >
          Start Game
        </button>
      </div>
    </div>
  );
};

StartFrame.propTypes = propTypes;

export default StartFrame;
