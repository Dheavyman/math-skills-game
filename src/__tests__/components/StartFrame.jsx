import React from 'react';
import TestRenderer from 'react-test-renderer';

import StartFrame from '../../components/StartFrame';

describe('Start frame component', () => {
  const props = {
    handleStartGame: jest.fn()
  };

  it('should render component successfully', () => {
    const renderer = TestRenderer.create(<StartFrame {...props} />);

    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
