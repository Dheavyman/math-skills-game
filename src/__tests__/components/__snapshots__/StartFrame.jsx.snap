// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Start frame component should render component successfully 1`] = `
<div
  className="row mt-5"
>
  <div
    className="col-12 text-center"
  >
    <button
      className="btn btn-lg btn-info"
      onClick={[MockFunction]}
      type="button"
    >
      Start Game
    </button>
  </div>
</div>
`;
